hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -326,13 +329,17 @@ public void setCurrentCompactionLog(long latestSequenceNum) {
     appendToCurrentCompactionLog("");
   }
 
+
   @Override
-  public void close() {
-    synchronized (this) {
-      if (!closed) {
-        closed = true;
-        if (executor != null) {
-          executor.shutdown();
+  public void close() throws Exception {
+    if (!closed) {
+      synchronized (this) {
+        if (!closed) {
+          closed = true;
```

Comment:
It would be better to change this flag after scheduler.close(), if you believe scheduler.close() throws some exception.

Reply:
[scheduler.close()](https://github.com/apache/ozone/blob/3964ece6561b0402165c4158d51b65e9d82c8860/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/Scheduler.java#L92) doesn't throw any exception. It swallows any exception and log it.